src/garage/replay_buffer/her_replay_buffer.py

```diff
@@ -97,3 +120,14 @@ def sample(self, batch_size):
             assert key in transitions, 'key %s missing from transitions' % key
 
         return transitions
+
+    def __getstate__(self):
```

Comment:
`__setstate__` is missing. When an instance is restored from a snapshot, there will be no `self._sample_transitions`.

Reply:
I think you need to store `replay_k, reward_fun` in `__getstate__` and regenerate a `_sample_transitions` in `__setstate__`

Reply:
Good catch